sample_information_validator.py

```diff
@@ -23,78 +23,280 @@
+            cell_value = str(self.work_sheet[cell_id].value)
+            if(cell_value.isspace()):
+                logger.warning(
+                    'Cell {} contains empty spaces only. Remove content.'.format(cell_id)
+                   )
+                global WARNINGS
+                WARNINGS += 1
+            elif(self.work_sheet[cell_id].value != None):
+                cellID_withSample.append(i)
+            else:
+                cellID_noSample.append(cell_id)# TODO check here that these rows do really not contain information
+        return(cellID_withSample)
+
+    def ProjectInfo(self, username_couchDB, password_couchDB, url):
+        url_string = "http://"+username_couchDB+":"+password_couchDB+url
+        connection = couchdb.Server(url=url_string)
+        db = connection["projects"]
+        # check the existence of the project number in couchDB
+        project_id_found = db.view("project/project_id", key=self.projectID())
+        prow=project_id_found.rows
+        # Project not found
+        if len(prow) == 0:
             logger.error(
-                    'Cell {} is not numeric'.format(cell.coordinate)
-                    )
-            return False
-        try:
-            float(cell.value)
-            return True
-        except ValueError:
+            'Project not found, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
+            )
+            quit()
+        # more than one project found
+        elif len(prow) > 1:
             logger.error(
-                    'Cell {} is numeric but cannot be '
-                    'transformed to float'.format(cell.coordinate)
+            'Project ID not unique, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
             )
-            return False
-        except TypeError:
-            if cell.value is None:
-                logger.warning(
-                    'Cell {} is numeric but empty'.format(cell.coordinate)
+            quit()
+        else:
+        # puts the Document of the identified project in a new variable "pdoc"
+            pdoc = db.get(prow[0].id)
+            return(pdoc)
+
+    def requirements(self, path):
+        if self.sample_rec == None:
+            wb_sample_recommendations = load_workbook(path, read_only=True, data_only=True)
+            self.sample_rec = wb_sample_recommendations['Numeric']
+
+    def prep_standards(self, info, recom_path):
+        ''' gets the sample requirements from the sample requirement excel sheet based
+            on the given sample prep type. '''
+
+        prep = info['details']['library_construction_method']
+
+        prep_recs = [None,None,None,None,None,None,None,None]
+        prep_type_found = False
+        for row in range(2, 15):
```

Comment:
The `15` here should also be a class variable I think. In the future when more preps are added, we need to update this number.

Reply:
True!